### PR TITLE
⛓ Add arbitrum reddit test chain

### PR DESCRIPTION
**Pull request by iskdrews**

### CODE REVIEW

File: packages/core/src/model/chain/redditArbitrum.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  multicall2Address: '',
+  rpcUrl: 'https://testnet.redditspace.com/rpc',
+  blockExplorerUrl: testnetRedditscanUrl,
+  getExplorerAddressLink: getAddressLink(testnetRedditscanUrl),
+  getExplorerTransactionLink: getTransactionLink(testnetRedditscanUrl),
+}
+
+export default {
+    RedditTestnet
+}
```
Comment (mj426382) [inline packages/core/src/model/chain/redditArbitrum.ts:21]: Please add end of line here

File: packages/core/src/model/chain/redditArbitrum.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const testnetRedditscanUrl = 'https://testnet.redditspace.com'
+
+export const RedditTestnet: Chain = {
+  chainId: 5391184,
+  chainName: 'RedditTestnet',
+  isTestChain: true,
+  isLocalChain: false,
+  multicallAddress: '0xFf79F208541DdF9dA6632dD851a269BbfA97d9aD',
+  multicall2Address: '',
```
Comment (mj426382) [inline packages/core/src/model/chain/redditArbitrum.ts:12]: Can you deploy multicall2?

File: packages/core/src/model/chain/redditArbitrum.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  rpcUrl: 'https://testnet.redditspace.com/rpc',
+  blockExplorerUrl: testnetRedditscanUrl,
+  getExplorerAddressLink: getAddressLink(testnetRedditscanUrl),
```
Comment (mj426382) [inline packages/core/src/model/chain/redditArbitrum.ts:15]: I think that something is broken with that, i pasted explorer address link with multicall address and there i had transactions with that address.
Either address link is wrong or multicall address, please check it.

File: packages/core/src/model/chain/redditArbitrum.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Chain } from '../../constants'
+import { getAddressLink, getTransactionLink } from '../../helpers/chainExplorerLink'
+
+const testnetRedditscanUrl = 'https://testnet.redditspace.com'
+
+export const RedditTestnet: Chain = {
```
Comment (mj426382) [inline packages/core/src/model/chain/redditArbitrum.ts:6]: Please consider rename into ArbitrumRedditTestnet

**Review by mj426382 — CHANGES_REQUESTED** [review]
PR looks good, but there are some weaknesses to change.

**Comment by mj426382** [comment]
Tests written by you do not pass, try to fix that and run pnpm test before pushing changes.

File: packages/core/src/model/chain/redditArbitrum.ts (hunk 2 above)
Comment (iskdrews) [reply packages/core/src/model/chain/redditArbitrum.ts:12]: Yup the multicall2 has been deployed at: `0xd4d664d419a6a845c98cc366ae1c4b24592bd5ce`

File: packages/core/src/model/chain/redditArbitrum.ts (hunk 3 above)
Comment (iskdrews) [reply packages/core/src/model/chain/redditArbitrum.ts:15]: I have fixed it, multicall: `0x722db82dea58c880d03b87885053f206f1b37136`

**Comment by iskdrews** [comment]
@mj426382 thanks for your comment, I have resolved all of them, and the tests are passing on my end. Could you review it again?

File: packages/core/src/model/chain/arbitrumReddit.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { Chain } from '../../constants'
+import { getAddressLink, getTransactionLink } from '../../helpers/chainExplorerLink'
+
+const testnetRedditscanUrl = 'https://testnet.redditspace.com'
```
[the remark this answers is not in the document]
Comment (iskdrews) [reply packages/core/src/model/chain/arbitrumReddit.ts]: Resolved.

File: packages/core/src/model/chain/arbitrumReddit.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+export const ArbitrumRedditTestnet: Chain = {
+  chainId: 5391184,
+  chainName: 'RedditTestnet',
```
[the remark this answers is not in the document]
Comment (iskdrews) [reply packages/core/src/model/chain/arbitrumReddit.ts]: Resolved.

File: packages/core/src/model/chain/arbitrumReddit.test.ts [hunk 7]
```diff
@@ -0,0 +1,33 @@
+import { expect } from 'chai'
+import { TEST_ADDRESS, TEST_TX } from './test-defaults'
+import { ArbitrumRedditTestnet } from '../..'
+
+describe('RedditTestnet Chain', () => {
```
Comment (mj426382) [inline packages/core/src/model/chain/arbitrumReddit.test.ts:5]:
```suggestion
describe('ArbitrumRedditTestnet Chain', () => {
```

**Review by mj426382 — CHANGES_REQUESTED** [review]
Can you verify multicall contracts code on https://testnet.redditspace.com/
There is only this small thing to merge this PR :)

**Comment by iskdrews** [comment]
> Can you verify multicall contracts code on https://testnet.redditspace.com/ There is only this small thing to merge this PR :)

They both are verified. 
- Multicall:   https://testnet.redditspace.com/address/0x722db82dea58c880D03b87885053F206f1B37136/contracts
- Multicall2: https://testnet.redditspace.com/address/0xD4D664D419A6A845C98Cc366ae1c4b24592BD5CE/contracts

**Comment by iskdrews** [comment]
@mj426382 btw I was trying to resolve this error by this PR: 
```
UnsupportedChainIdError: Unsupported chain id: 5391184. Supported chain ids are: 1337,31337,43114,43113,42161,421611,1313161554,1313161555,1,3,4,5,42,56,97,25,338,250,4002,100,1666600000,1088,588,1285,1287,1284,11297108109,11297108099,137,80001,42262,42261,19,361,365,108,18,69,10,106,111,280.
    at useEthers (useEthers.ts:67)
    at Component (HeaderWarning.js:10)
    at renderWithHooks (react-dom.development.js:14985)
    at updateFunctionComponent (react-dom.development.js:17356)
    at beginWork (react-dom.development.js:19063)
    at beginWork$1 (react-dom.development.js:23940)
    at performUnitOfWork (react-dom.development.js:22776)
    at workLoopSync (react-dom.development.js:22707)
    at renderRootSync (react-dom.development.js:22670)
    at performSyncWorkOnRoot (react-dom.development.js:22293)
    at react-dom.development.js:11327
    at unstable_runWithPriority (scheduler.development.js:468)
    at runWithPriority$1 (react-dom.development.js:11276)
    at flushSyncCallbackQueueImpl (react-dom.development.js:11322)
    at flushSyncCallbackQueue (react-dom.development.js:11309)
    at flushSync (react-dom.development.js:22467)
    at Object.scheduleRefresh (react-dom.development.js:24429)
    at react-refresh-runtime.development.js:304
    at Set.forEach (<anonymous>)
    at Object.performReactRefresh (react-refresh-runtime.development.js:293)
    at RefreshUtils.js:62 5391184 "ERROR!!!!!!!!!!"
```

**Review by mj426382 — CHANGES_REQUESTED** [review]
Everything looks fine, run `pnpm lint:fix` in core folder and update changes.

**Comment by iskdrews** [comment]
> Everything looks fine, please run `pnpm lint:fix` in core folder and update changes.
@mj426382 done.

**Review by mj426382 — CHANGES_REQUESTED** [review]
Just one little thing.
Please refactor one test which doesn't pass.
Run `pnpm test` and then refactor it.

**Comment by iskdrews** [comment]
@mj426382 please which test you mean is failing, I checked the testing logs here for the ArbitrumReddit test script and it appears that it's working successfully. 
Screenshot from 2022-06-18 15-30-09

**Comment by iskdrews** [comment]
@mj426382 do you mean this failing test? 
```
 |   408 passing (1m)
 |   2 pending
 |   1 failing
 | 
 |   1) useConfig
 |        default testing config:
 | 
 |       AssertionError: expected 43 to equal 42
 |       + expected - actual
 | 
 |       -43
 |       +42
 |       
 |       at Context.<anonymous> (src/hooks/useConfig.test.tsx:47:112)
 |       at runMicrotasks (<anonymous>)
 |       at processTicksAndRejections (node:internal/process/task_queues:96:5)
 | 
 | 
 | 
 | error Command failed with exit code 1.
 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
 | `yarn test -- --timeout 0` failed with exit code 1
Stopping 3 active children
```

**Comment by mj426382** [comment]
> @mj426382 do you mean this failing test?
> 
> ```
>  |   408 passing (1m)
>  |   2 pending
>  |   1 failing
>  | 
>  |   1) useConfig
>  |        default testing config:
>  | 
>  |       AssertionError: expected 43 to equal 42
>  |       + expected - actual
>  | 
>  |       -43
>  |       +42
>  |       
>  |       at Context.<anonymous> (src/hooks/useConfig.test.tsx:47:112)
>  |       at runMicrotasks (<anonymous>)
>  |       at processTicksAndRejections (node:internal/process/task_queues:96:5)
>  | 
>  | 
>  | 
>  | error Command failed with exit code 1.
>  | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
>  | `yarn test -- --timeout 0` failed with exit code 1
> Stopping 3 active children
> ```

yes, because you added reddis chain to supported, so the number was increased by 1

**Review by mj426382 — APPROVED** [review]

**Pull request merged**